File: PlayWall/src/main/scala/de/tobias/playpad/action/actions/cart/handler/CartActionHandlerFactory.scala
package de.tobias.playpad.action.actions.cart.handler

import de.tobias.playpad.action.actions.cart.CartAction.CartActionMode

object CartActionHandlerFactory {

	def getInstance(cartActionMode: CartActionMode): CartActionHandler = {
		cartActionMode match {
			case CartActionMode.PLAY_STOP => new PlayStopHandler()
			case CartActionMode.PLAY_PAUSE => new PlayPauseHandler()
			case CartActionMode.PLAY_HOLD => new PlayHoldHandler()
			case CartActionMode.PLAY_PLAY => new PlayPlayHandler()
		}
	}
}

File: PlayWall/src/main/java/de/tobias/playpad/action/actions/cart/handler/CartActionHandlerFactory.java
package de.tobias.playpad.action.actions.cart.handler;

import de.tobias.playpad.action.actions.CartAction;

public class CartActionHandlerFactory
{
	private CartActionHandlerFactory()
	{
	}

	public static CartActionHandler getInstance(CartAction.CartActionMode cartActionMode)
	{
		switch(cartActionMode)
		{
			case PLAY_PAUSE:
				return new PlayPauseHandler();
			case PLAY_STOP:
				return new PlayStopHandler();
			case PLAY_HOLD:
				return new PlayHoldHandler();
			case PLAY_PLAY:
				return new PlayPlayHandler();
			default:
				return null;
		}
	}
}
